src/sage/matrix/matrix_integer_dense.pyx

```
@@ -3177,6 +3202,21 @@ cdef class Matrix_integer_dense(Matrix_dense):
             raise TypeError("precision prec must be >= 0")
         prec = int(prec)
 
+        if algorithm == 'flatter':
+            import subprocess
+            cmd = ["flatter"]
+            if fp is not None or early_red or use_givens or transformation or eta is not None or use_siegel:
+                raise TypeError("flatter does not support fp, early_red, use_givens, transformation, eta or use_siegel")
+            if kwds:
+                raise TypeError("flatter does not support additional keywords")
```

would it be more consistent to just ignore `kwds` here?

---

I think it's better practice to complain to the user about invalid data than to silently ignore them.